src/combinatorics/simple_graph/inc_matrix.lean

```diff
@@ -0,0 +1,192 @@
+/-
+Copyright (c) 2021 Gabriel Moise. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gabriel Moise, Yaël Dillies
+-/
+import combinatorics.simple_graph.basic
+import data.matrix.basic
+
+/-!
+# Incidence matrix of a simple graph
+
+This file defines the unoriented incidence matrix of a simple graph.
+
+## Main definitions
+
+* `simple_graph.inc_matrix`: `G.inc_matrix R` is the incidence matrix of `G` over the ring `R`.
+
+## Main results
+
+* `simple_graph.inc_matrix_mul_transpose_diag`: The sum of the `a`-th row of `G.inc_matrix R`
+  equals the degree of vertex `a`.
```

Comment:
It seems like it would be worth mentioning this one:
```
* `simple_graph.inc_matrix_mul_transpose`: Gives a complete description of the product of
`G.inc_matrix R` and its transpose; the diagonal is the degrees of each vertex, and the
off-diagonals are 1 or 0 depending on whether or not the vertices are adjacent.
```